### PR TITLE
[4.3.4] En-1913-434-digital-assets-update-without-file-delete-saved-metadata

### PR DESCRIPTION
**Pull request by alepintus**
Hi @joewhite101 @eugeniosant ,
Saving metadata when updating an existing resource fixed.
Could you please review this PR?

Thanks!

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

File: admin-console/src/main/java/com/agiletec/plugins/jacms/apsadmin/resource/MultipleResourceAction.java [hunk 1]
```diff
@@ -201,7 +214,22 @@ public String save() {
                     resourceFile.setMainGroup(getMainGroup());
                     resourceFile.setResourceType(this.getResourceType());
                     resourceFile.setCategories(getCategories());
-                    resourceFile.setMetadata(imgMetadata);
+                    logger.debug("this.getStrutsAction() {}", this.getStrutsAction());
+
+                    if (ApsAdminSystemConstants.EDIT == this.getStrutsAction()) {
```
Comment (joewhite101) [inline admin-console/src/main/java/com/agiletec/plugins/jacms/apsadmin/resource/MultipleResourceAction.java]: Does this if{} branch have some test coverage?

File: admin-console/src/main/java/com/agiletec/plugins/jacms/apsadmin/resource/MultipleResourceAction.java [hunk 2]
```diff
@@ -137,15 +137,28 @@ protected void validateFileDescriptions() {
...
+        fetchMetadataEdit();
         return super.removeCategory();
     }
 
+    protected void fetchMetadataEdit() {
+        if (ApsAdminSystemConstants.EDIT == this.getStrutsAction()) {
+            try {
+                this.setMetadata(this.loadResource(this.getResourceId()).getMetadata());
+            } catch (Throwable ex) {
+                logger.debug("error reading resource on fetchMetadataEdit {}", ex);
```
Comment (joewhite101) [inline admin-console/src/main/java/com/agiletec/plugins/jacms/apsadmin/resource/MultipleResourceAction.java]: This should be logged at error

File: admin-console/src/main/java/com/agiletec/plugins/jacms/apsadmin/resource/MultipleResourceAction.java [hunk 3]
```diff
@@ -137,15 +137,28 @@ protected void validateFileDescriptions() {
...
+            }
+            logger.debug("metadata size: {}", getMetadata().size());
```
Comment (joewhite101) [inline admin-console/src/main/java/com/agiletec/plugins/jacms/apsadmin/resource/MultipleResourceAction.java]: Would it be possible to add some identifying issue to this message so we know which asset it applies to?

**Comment by alepintus** [comment]
Hey @joewhite101,
thanks for the review!
Log fixed and test added.

Thanks!

**Comment by entando-service** [comment]
All is well

**Pull request merged**